src/cdk/overlay/overlay-ref.ts

```diff
@@ -381,7 +385,9 @@ export class OverlayRef implements PortalOutlet, OverlayReference {
 
     // Forward backdrop clicks such that the consumer of the overlay can perform whatever
     // action desired when such a click occurs (usually closing the overlay).
-    this._backdropElement.addEventListener('click', this._backdropClickHandler);
+    this._ngZone.runOutsideAngular(() => {
```

Comment:
Could this change be decoupled from the length check above? I suspect we'll run into a number of apps that depend on change detection from this event

Reply:
Hey, could you elaborate a little bit on that? E.g. some code example?

Reply:
It's not something it's easy to create a code example for, unfortunately. We often run into a problem when landing changes like this into Google where some team's unit tests are implicitly depending on specific number and sequence of change detections in order for their assertions to pass. Removing change detection from something that used to have one almost always makes some number of tests fail.